src/utils/workflowModifier.js

```diff
@@ -0,0 +1,31 @@
+const _ = require('lodash');
+
+async function removeNodesByCategory(blueprint_spec, category) {
+  for (const node of blueprint_spec.nodes) {
+    if (node.next && typeof node.next !== 'object') {
+      let nextNode = node;
+      do {
```

Comment:
achei que vc deu uma volta aqui que não precisava...
se a category do meu next for a para remoção, aponto para o next dele, senão sigo para o próximo.
```
const nextNode = find(id === myNode.next)
if(nextNode.category === category) then (myNode.next = nextNode.next)
```

inclusive essa função de substituição do next é uma _commom function_ que vc evoca aqui e também no tratamento do flow... poderia isolar em uma função separada e reaproveitar o código.

Reply:
No caso, coloquei o do.. while para casos de nós de setToBag seguidos.. De fato posso tentar colocar isso em uma function e apontar pra ela em casos de flow ou não